pkg/agent/cluster/controllers.go

```diff
@@ -0,0 +1,42 @@
+package cluster
+
+import (
+	"context"
+
+	clusterController "github.com/rancher/rancher/pkg/controllers/user"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+func RunControllers() error {
+	logrus.Info("Starting user controllers")
+	c, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	scaledContext, err := config.NewScaledContext(*c)
```

Comment:
I'd prefer we have `config.NewUserOnlyContext(*c)`.  Even if it does all this same stuff you have here, it's just the encapsulation.  It seem wrong to be creating a "scaledcontext" and "usercontext" to just get the useronlycontext here.